packages/protocol/contracts/governance/Governance.sol

```diff
@@ -631,26 +635,117 @@ contract Governance is
-      voteRecord.weight,
-      weight,
-      (voteRecord.proposalId == proposalId) ? voteRecord.value : Proposals.VoteValue.None,
-      value
+
+    Proposals.VoteValue[] memory currentValues = new Proposals.VoteValue[](1);
+    currentValues[0] = value;
+    uint256[] memory currentWeights = new uint256[](1);
+    currentWeights[0] = weight;
+
+    voteInternal(proposal, proposalId, index, weight, account, currentValues, currentWeights);
+    return true;
+  }
+
+  /**
+   * @notice Votes partially on a proposal in the referendum stage.
+   * @param proposalId The ID of the proposal to vote on.
+   * @param index The index of the proposal ID in `dequeued`.
+   * @param voteValues Whether to vote yes, no, or abstain.
+   * @param weights Weights of vote choices.
+   * @return Whether or not the vote was cast successfully.
+   */
+  /* solhint-disable code-complexity */
+  function votePartially(
+    uint256 proposalId,
+    uint256 index,
+    Proposals.VoteValue[] calldata voteValues,
+    uint256[] calldata weights
+  ) external nonReentrant returns (bool) {
+    require(voteValues.length == weights.length, "Incorrect length");
+    require(voteValues.length <= 3, "Only values from VoteValue allowed");
+
+    dequeueProposalsIfReady();
+    (Proposals.Proposal storage proposal, Proposals.Stage stage) = requireDequeuedAndDeleteExpired(
+      proposalId,
+      index
+    );
+    if (!proposal.exists()) {
+      return false;
+    }
+    require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
+
+    address account = getAccounts().voteSignerToAccount(msg.sender);
+    uint256 totalLockedGold = getLockedGold().getAccountTotalLockedGold(account);
+    voteInternal(proposal, proposalId, index, totalLockedGold, account, voteValues, weights);
+
+    return true;
+  }
+
+  /**
+   * @notice Votes on a proposal in the referendum stage.
+   * @param proposal The proposal struct.
+   * @param proposalId The ID of the proposal to vote on.
+   * @param index The index of the proposal ID in `dequeued`.
+   * @param totalLockedGold Total locked gold owned by account.
+   * @param account Account based on signer.
+   * @param voteValues Whether to vote yes, no, or abstain.
+   * @param weights Weights of vote choices.
+   * @return Whether or not the proposal is passing.
+   */
+  function voteInternal(
+    Proposals.Proposal storage proposal,
+    uint256 proposalId,
+    uint256 index,
+    uint256 totalLockedGold,
+    address account,
+    Proposals.VoteValue[] memory voteValues,
+    uint256[] memory weights
+  ) private {
+    Voter storage voter = voters[account];
+
+    require(
+      totalLockedGold >= getTotalWeightRequested(weights),
+      "Voter doesn't have enough locked gold"
     );
+    VoteRecord storage previousVoteRecord = voter.referendumVotes[index];
+
+    if (previousVoteRecord.proposalId != proposalId) {
+      proposal.updateVote(new Proposals.VoteValue[](0), new uint256[](0), voteValues, weights);
```

Comment:
Can you add a comment about what is this case? Why the `proposalId`wouldn't match?